src/nodejs/unit-http/http_server.js

```diff
@@ -413,7 +414,14 @@ ServerRequest.prototype._read = function _read(n) {
 };
 
 
-function Server(requestListener) {
+function Server(options, requestListener) {
+    if (typeof options === 'function') {
+        requestListener = options;
+        options = {};
+    } else {
```

Comment:
This logic needs a little adjustment since an error message will be logged even when the `options` are an object. I would suggest adding an additional check in tests to verify this (please see below).

As far as I understood, NodeJS is also happy with `null` here, so we can copy this behavior too (to reduce further rework here).

Reply:
This check is a slight adjustment of what the nodejs source does: https://github.com/nodejs/node/pull/15752/files#diff-2464899ab38c61dd42654fdb9658c75989cb6883b9c9d76613872a12d3fe2780R273-R278

The idea there is to figure out whether people are calling this function:

```js
function Server(options, requestListener) {...}
```
with one, or two arguments:

```js
Server(listener); // one argument

Server(options, listener); // two arguments
```

The only thing we care about right now is that if the first argument is a function, then we assume the function was called without the options argument, and the first incoming arg as `option` is actually intended to be a request listener.

If the users called this function with two arguments (ie the first argument is not a function), then it doesn't matter what the options is (undefined, null, empty object, actual object...), we should log that we are aware there's something, we're just not doing anything about it at the moment.

I don't think this needs changing.

Reply:
> If the users called this function with two arguments (ie the first argument is not a function), then it doesn't matter what the options is (undefined, null, empty object, actual object...)

yeah, in this case no changes in logic required. I just assumed that sinse `{}` and `null` are valid values we could not report error in this case. Please ignore this one.